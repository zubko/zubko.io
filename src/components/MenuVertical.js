/**
 * Vertical top bar menu component. It's displayed on small screens.
 * @flow
 */

import React, { Fragment } from 'react';

import { Link } from '../components';
import { Colors } from '../Theme';

const Separator = () => <span css={Styles.separator}>{' â€“ '}</span>;

type Props = {
  links: { path: string, title: string }[],
  isHead: boolean,
  isBody: boolean,
  isOpened: boolean,
  setOpened: (val: boolean) => any,
  css: any,
};

const MenuVertical = ({
  isHead,
  isBody,
  isOpened,
  setOpened,
  links,
  ...otherProps
}: Props) => {
  return (
    <div {...otherProps}>
      {isHead ? (
        <button
          css={[Styles.button, isOpened ? Styles.buttonSelected : null]}
          onClick={() => setOpened(!isOpened)}>
          Menu
        </button>
      ) : null}
      {isBody ? (
        <div
          css={[
            Styles.body,
            !isOpened
              ? {
                  display: 'none',
                }
              : null,
          ]}>
          {links.map((item, index) => (
            <Fragment key={item.path}>
              <Link to={item.path} css={Styles.menuItem}>
                {item.title}
              </Link>
              {index < links.length - 1 ? <Separator /> : null}
            </Fragment>
          ))}
        </div>
      ) : null}
    </div>
  );
};

export default MenuVertical;

const Styles = {
  button: {
    cursor: 'pointer',
    color: 'white',
    border: '1px solid white',
    padding: '0.5rem',
    backgroundColor: 'transparent',
    ':focus': { outline: 0 },
  },
  buttonSelected: {
    backgroundColor: Colors.mainDarker,
  },
  body: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    paddingBottom: '1rem',
  },
  menuItem: {
    color: 'white',
    textDecoration: 'none',
    ':hover': {
      color: `white`,
    },
    fontSize: '1rem',
    fontWeight: 'medium',
  },
  separator: {
    color: 'white',
    cursor: 'default',
  },
};
