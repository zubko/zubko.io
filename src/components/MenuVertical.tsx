/**
 * Vertical top bar menu component. It's displayed on small screens.

 */

import { Fragment } from 'react';
import { Colors } from '../Theme';
import { rhythm } from '../Typography';
import Link from './Link';

const Separator = () => <span css={Styles.separator}> </span>;

type Props = {
  links: { path: string, title: string }[],
  isHead: boolean,
  isBody: boolean,
  isOpened: boolean,
  setOpened: (val: boolean) => any,
};

const MenuVertical = ({
  isHead,
  isBody,
  isOpened,
  setOpened,
  links,
  ...otherProps
}: Props) => {
  return (
    <div {...otherProps}>
      {isHead ? (
        <button
          css={[Styles.button, isOpened ? Styles.buttonSelected : null]}
          onClick={() => setOpened(!isOpened)}>
          Menu
        </button>
      ) : null}
      {isBody ? (
        <div
          css={[
            Styles.body,
            !isOpened
              ? {
                  display: 'none',
                }
              : null,
          ]}>
          {links.map((item, index) => (
            <Fragment key={item.path}>
              <Link usePlainStyle to={item.path} css={Styles.menuLink}>
                {item.title}
              </Link>
              {index < links.length - 1 ? <Separator /> : null}
            </Fragment>
          ))}
        </div>
      ) : null}
    </div>
  );
};

export default MenuVertical;

const Styles = {
  button: {
    cursor: 'pointer',
    color: Colors.mainDarker,
    border: `1px solid ${Colors.mainDarker}`,
    padding: '0.5rem',
    backgroundColor: 'transparent',
    ':focus': { outline: 0 },
  },
  buttonSelected: {
    color: 'white',
    border: `1px solid white`,
    backgroundColor: Colors.mainDarker,
    background: `repeating-linear-gradient(
      45deg,
      ${Colors.main},
      ${Colors.main} 10px,
      ${Colors.mainDarker} 10px,
      ${Colors.mainDarker} 20px
    )`,
    opacity: 0.9,
  },
  body: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    paddingBottom: rhythm(0.5),
    paddingTop: rhythm(1),
  },
  menuLink: {
    textDecoration: 'none',
    ':hover': {},
    fontSize: '1rem',
    fontWeight: 'medium',
    background: 'none',
  },
  separator: {
    height: '0.5rem',
    color: 'white',
    cursor: 'default',
  },
};
