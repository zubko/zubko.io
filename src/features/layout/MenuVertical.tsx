/**
 * Vertical top bar menu component. It's displayed on small screens.
 */

import { Fragment } from "react";
import styled from "styled-components";
import { Link } from "../../components/Link";
import { Colors } from "../../Theme";
import { rhythm } from "../../Typography";

type Props = {
  className?: string;
  links: { path: string; title: string }[];
  isHead?: boolean;
  isBody?: boolean;
  isOpened: boolean;
  onOpened?: (val: boolean) => void;
};

export const MenuVertical = ({
  isHead,
  isBody,
  isOpened,
  onOpened,
  links,
  ...otherProps
}: Props) => {
  return (
    <div {...otherProps}>
      {isHead ? (
        <MenuButton isOpened={isOpened} onClick={() => onOpened?.(!isOpened)}>
          Menu
        </MenuButton>
      ) : null}
      {isBody ? (
        <Body isOpened={isOpened}>
          {links.map((item, index) => (
            <Fragment key={item.path}>
              <MenuLink usePlainStyle to={item.path}>
                {item.title}
              </MenuLink>
              {index < links.length - 1 ? <Separator /> : null}
            </Fragment>
          ))}
        </Body>
      ) : null}
    </div>
  );
};

const MenuButton = styled("button")<{ isOpened: boolean }>(({ isOpened }) => [
  `
  cursor: pointer;
  color: ${Colors.mainDarker};
  border: 1px solid ${Colors.mainDarker};
  padding: 0.5rem;
  background-color: transparent;
  :focus {
    outline: 0;
  }
`,
  isOpened
    ? `
  color: white;
  border: 1px solid white;
  background-color: ${Colors.mainDarker};
  background: repeating-linear-gradient(
    45deg,
    ${Colors.main},
    ${Colors.main} 10px,
    ${Colors.mainDarker} 10px,
    ${Colors.mainDarker} 20px
  );
  opacity: 0.9;
`
    : null,
]);

const Separator = styled.span`
  height: 0.5rem;
  color: white;
  cursor: default;
`;

const Body = styled.div<{ isOpened: boolean }>`
  display: ${({ isOpened }) => (isOpened ? "flex" : "none")};
  flex-direction: column;
  align-items: center;
  padding-bottom: ${rhythm(0.5)};
  padding-top: ${rhythm(1)};
`;

const MenuLink = styled(Link)`
  text-decoration: none;
  :hover: {
  }
  font-size: 1rem;
  font-weight: medium;
  background: none;
`;
