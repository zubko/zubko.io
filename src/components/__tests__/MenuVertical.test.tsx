import React from "react";
import renderer from "react-test-renderer";

import MenuVertical from "../MenuVertical";

describe("MenuVertical", () => {
  const links = [
    { path: "/path-1", title: "title-1" },
    { path: "/path-2", title: "title-2" },
  ];

  it("head renders without regressions", () => {
    const tree = renderer
      .create(<MenuVertical isHead isOpened={false} links={links} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("body renders without regressions", () => {
    const tree = renderer
      .create(<MenuVertical isBody isOpened={false} links={links} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("opened head renders without regressions", () => {
    const tree = renderer
      .create(<MenuVertical isHead isOpened links={links} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("opened body renders without regressions", () => {
    const tree = renderer
      .create(<MenuVertical isBody isOpened links={links} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
