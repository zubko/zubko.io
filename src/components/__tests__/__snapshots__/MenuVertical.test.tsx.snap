// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuVertical body renders without regressions 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1tobc0i-MenuVertical{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-bottom:0.809rem;padding-top:1.618rem;display:none;}",
      }
    }
    data-emotion="css 1tobc0i-MenuVertical"
  />
  <div
    className="css-1tobc0i-MenuVertical"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1mgyam6{-webkit-text-decoration:none;text-decoration:none;font-size:1rem;font-weight:medium;background:none;}",
        }
      }
      data-emotion="css 1mgyam6"
    />
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-t9hu3x{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;font-size:1rem;font-weight:medium;background:none;}.css-t9hu3x:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}",
        }
      }
      data-emotion="css t9hu3x"
    />
    <a
      className="css-t9hu3x"
      href="/path-1"
    >
      title-1
    </a>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-l1p1i0-Separator{height:0.5rem;color:white;cursor:default;}",
        }
      }
      data-emotion="css l1p1i0-Separator"
    />
    <span
      className="css-l1p1i0-Separator"
    >
       
    </span>
    <a
      className="css-t9hu3x"
      href="/path-2"
    >
      title-2
    </a>
  </div>
</div>
`;

exports[`MenuVertical head renders without regressions 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-50n0xx-MenuVertical{cursor:pointer;color:#2C8127;border:1px solid #2C8127;padding:0.5rem;background-color:transparent;}.css-50n0xx-MenuVertical:focus{outline:0;}",
      }
    }
    data-emotion="css 50n0xx-MenuVertical"
  />
  <button
    className="css-50n0xx-MenuVertical"
    onClick={[Function]}
  >
    Menu
  </button>
</div>
`;

exports[`MenuVertical opened body renders without regressions 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1icg5sk-MenuVertical{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-bottom:0.809rem;padding-top:1.618rem;}",
      }
    }
    data-emotion="css 1icg5sk-MenuVertical"
  />
  <div
    className="css-1icg5sk-MenuVertical"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1mgyam6{-webkit-text-decoration:none;text-decoration:none;font-size:1rem;font-weight:medium;background:none;}",
        }
      }
      data-emotion="css 1mgyam6"
    />
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-t9hu3x{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;font-size:1rem;font-weight:medium;background:none;}.css-t9hu3x:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}",
        }
      }
      data-emotion="css t9hu3x"
    />
    <a
      className="css-t9hu3x"
      href="/path-1"
    >
      title-1
    </a>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-l1p1i0-Separator{height:0.5rem;color:white;cursor:default;}",
        }
      }
      data-emotion="css l1p1i0-Separator"
    />
    <span
      className="css-l1p1i0-Separator"
    >
       
    </span>
    <a
      className="css-t9hu3x"
      href="/path-2"
    >
      title-2
    </a>
  </div>
</div>
`;

exports[`MenuVertical opened head renders without regressions 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-nf73zk-MenuVertical{cursor:pointer;color:#2C8127;border:1px solid #2C8127;padding:0.5rem;background-color:transparent;color:white;border:1px solid white;background-color:#2C8127;background:repeating-linear-gradient(
      45deg,
      #3CB534,
      #3CB534 10px,
      #2C8127 10px,
      #2C8127 20px
    );opacity:0.9;}.css-nf73zk-MenuVertical:focus{outline:0;}",
      }
    }
    data-emotion="css nf73zk-MenuVertical"
  />
  <button
    className="css-nf73zk-MenuVertical"
    onClick={[Function]}
  >
    Menu
  </button>
</div>
`;
